[2024-11-12T14:36:23.262+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-12T14:36:23.279+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T14:36:23.290+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T14:36:23.291+0000] {taskinstance.py:2865} INFO - Starting attempt 3 of 6
[2024-11-12T14:36:23.307+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark-submit> on 2024-11-11 00:00:00+00:00
[2024-11-12T14:36:23.320+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'api_test_v1', 'spark-submit', 'scheduled__2024-11-11T00:00:00+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/Weather_etl_dag.py', '--cfg-path', '/tmp/tmpk5xhyz72']
[2024-11-12T14:36:23.324+0000] {standard_task_runner.py:105} INFO - Job 7: Subtask spark-submit
[2024-11-12T14:36:23.328+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=644) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-11-12T14:36:23.330+0000] {standard_task_runner.py:72} INFO - Started process 645 to run task
[2024-11-12T14:36:23.379+0000] {task_command.py:467} INFO - Running <TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [running]> on host 3eb42027ba35
[2024-11-12T14:36:23.465+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Simo' AIRFLOW_CTX_DAG_ID='api_test_v1' AIRFLOW_CTX_TASK_ID='spark-submit' AIRFLOW_CTX_EXECUTION_DATE='2024-11-11T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-11T00:00:00+00:00'
[2024-11-12T14:36:23.467+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-12T14:36:23.504+0000] {spark_submit.py:304} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-11-12T14:36:23.506+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark /dags/pysparksubmitoperator.py
[2024-11-12T14:36:23.579+0000] {spark_submit.py:634} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-12T14:36:25.999+0000] {spark_submit.py:634} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-11-12T14:36:26.000+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2024-11-12T14:36:26.001+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2024-11-12T14:36:26.002+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2024-11-12T14:36:26.003+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2024-11-12T14:36:26.004+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1103)
[2024-11-12T14:36:26.005+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1103)
[2024-11-12T14:36:26.005+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2024-11-12T14:36:26.006+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1120)
[2024-11-12T14:36:26.007+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1129)
[2024-11-12T14:36:26.008+0000] {spark_submit.py:634} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-11-12T14:36:26.030+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark /dags/pysparksubmitoperator.py. Error code is: 1.
[2024-11-12T14:36:26.049+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=api_test_v1, task_id=spark-submit, run_id=scheduled__2024-11-11T00:00:00+00:00, execution_date=20241111T000000, start_date=20241112T143623, end_date=20241112T143626
[2024-11-12T14:36:26.087+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-12T14:36:26.089+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 7 for task spark-submit (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark /dags/pysparksubmitoperator.py. Error code is: 1.; 645)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark /dags/pysparksubmitoperator.py. Error code is: 1.
[2024-11-12T14:36:26.124+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-12T14:36:26.154+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-12T14:36:26.158+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-11-12T18:37:37.420+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-12T18:37:37.439+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T18:37:37.453+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T18:37:37.454+0000] {taskinstance.py:2865} INFO - Starting attempt 3 of 6
[2024-11-12T18:37:37.473+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark-submit> on 2024-11-11 00:00:00+00:00
[2024-11-12T18:37:37.481+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'api_test_v1', 'spark-submit', 'scheduled__2024-11-11T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/Weather_etl_dag.py', '--cfg-path', '/tmp/tmp2jmpl2eo']
[2024-11-12T18:37:37.485+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark-submit
[2024-11-12T18:37:37.488+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=709) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-11-12T18:37:37.490+0000] {standard_task_runner.py:72} INFO - Started process 711 to run task
[2024-11-12T18:37:37.590+0000] {task_command.py:467} INFO - Running <TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [running]> on host 287ce275bbd8
[2024-11-12T18:37:37.692+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Simo' AIRFLOW_CTX_DAG_ID='api_test_v1' AIRFLOW_CTX_TASK_ID='spark-submit' AIRFLOW_CTX_EXECUTION_DATE='2024-11-11T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-11T00:00:00+00:00'
[2024-11-12T18:37:37.695+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-12T18:37:37.736+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-11-12T18:37:37.740+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py
[2024-11-12T18:37:37.793+0000] {spark_submit.py:634} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-12T18:37:40.290+0000] {spark_submit.py:634} INFO - python3: can't open file '/jobs/pysparksubmitoperator.py': [Errno 2] No such file or directory
[2024-11-12T18:37:40.309+0000] {spark_submit.py:634} INFO - 24/11/12 18:37:40 INFO ShutdownHookManager: Shutdown hook called
[2024-11-12T18:37:40.315+0000] {spark_submit.py:634} INFO - 24/11/12 18:37:40 INFO ShutdownHookManager: Deleting directory /tmp/spark-718e7104-c255-4438-b62b-d686dd7e3452
[2024-11-12T18:37:40.361+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.
[2024-11-12T18:37:40.375+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=api_test_v1, task_id=spark-submit, run_id=scheduled__2024-11-11T00:00:00+00:00, execution_date=20241111T000000, start_date=20241112T183737, end_date=20241112T183740
[2024-11-12T18:37:40.403+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-12T18:37:40.405+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 8 for task spark-submit (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.; 711)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.
[2024-11-12T18:37:40.442+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-12T18:37:40.563+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-12T18:37:40.566+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-11-12T19:00:11.312+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-12T19:00:11.331+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T19:00:11.340+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [queued]>
[2024-11-12T19:00:11.341+0000] {taskinstance.py:2865} INFO - Starting attempt 3 of 6
[2024-11-12T19:00:11.356+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark-submit> on 2024-11-11 00:00:00+00:00
[2024-11-12T19:00:11.364+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'api_test_v1', 'spark-submit', 'scheduled__2024-11-11T00:00:00+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/Weather_etl_dag.py', '--cfg-path', '/tmp/tmpy1lyz6xn']
[2024-11-12T19:00:11.367+0000] {standard_task_runner.py:105} INFO - Job 7: Subtask spark-submit
[2024-11-12T19:00:11.370+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=697) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-11-12T19:00:11.371+0000] {standard_task_runner.py:72} INFO - Started process 698 to run task
[2024-11-12T19:00:11.419+0000] {task_command.py:467} INFO - Running <TaskInstance: api_test_v1.spark-submit scheduled__2024-11-11T00:00:00+00:00 [running]> on host c561c85cf2e2
[2024-11-12T19:00:11.494+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Simo' AIRFLOW_CTX_DAG_ID='api_test_v1' AIRFLOW_CTX_TASK_ID='spark-submit' AIRFLOW_CTX_EXECUTION_DATE='2024-11-11T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-11T00:00:00+00:00'
[2024-11-12T19:00:11.495+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-12T19:00:11.518+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-11-12T19:00:11.520+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py
[2024-11-12T19:00:11.565+0000] {spark_submit.py:634} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-12T19:00:13.649+0000] {spark_submit.py:634} INFO - python3: can't open file '/jobs/pysparksubmitoperator.py': [Errno 2] No such file or directory
[2024-11-12T19:00:13.663+0000] {spark_submit.py:634} INFO - 24/11/12 19:00:13 INFO ShutdownHookManager: Shutdown hook called
[2024-11-12T19:00:13.666+0000] {spark_submit.py:634} INFO - 24/11/12 19:00:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-9f28bed6-37c1-475f-a787-dce78aad78a2
[2024-11-12T19:00:13.707+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.
[2024-11-12T19:00:13.717+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=api_test_v1, task_id=spark-submit, run_id=scheduled__2024-11-11T00:00:00+00:00, execution_date=20241111T000000, start_date=20241112T190011, end_date=20241112T190013
[2024-11-12T19:00:13.734+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-12T19:00:13.735+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 7 for task spark-submit (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.; 698)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /jobs/pysparksubmitoperator.py. Error code is: 2.
[2024-11-12T19:00:13.760+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-12T19:00:13.778+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-12T19:00:13.781+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
